source/module_esolver/esolver_ks.cpp

```diff
@@ -379,6 +379,60 @@ void ESolver_KS<T, Device>::hamilt2density(const int istep, const int iter, cons
     ModuleBase::timer::tick(this->classname, "hamilt2density");
 }
 
+template <typename T, typename Device>
+void ESolver_KS<T, Device>::diag(const int istep, const int iter, const double ethr)
```

Comment:
rename 'diag' as 'hamilt2density', before this, rename the original 'hamilt2density' as 'hamilt2density_single'